ocaml/xapi/xapi_host.ml

```diff
@@ -1400,10 +1399,16 @@ let install_server_certificate ~__context ~host ~certificate ~private_key
   Xapi_mgmt_iface.reconfigure_stunnel ~__context
 
 let emergency_reset_server_certificate ~__context =
-  let args =
-    [!Xapi_globs.server_cert_path; Helper_hostname.get_hostname (); "--force"]
+  let xapi_ssl_pem = !Xapi_globs.server_cert_path in
+  let common_name, alt_names =
+    match Gencertlib.Lib.hostnames () with
+    | cn :: alt ->
+        (cn, alt)
+    | [] ->
+        (Helper_hostname.get_hostname (), [])
+    (* should never happen *)
   in
-  ignore @@ Gencert.call_generate_ssl_cert ~args ;
+  Gencertlib.Selfcert.host common_name alt_names xapi_ssl_pem ;
```

Comment:
I think we always want to include the main name in the subject alt names.
The CN field is deprecated in X509 certs, the subject-alt-names is supposed to be used instead, so this should always contain the full list of hostnames

Reply:
also https://tools.ietf.org/html/rfc6125#section-6.4.4

Reply:
Ok - so don't split the names - use the full list in alt_names?

Reply:
I agree with adding all hostnames into the SANs.

I'm not opposed to removing the CN from the certificate: it would not affect RFC-compliant clients, but I think it's worth communicating to teams building xapi clients they will not be able to rely on its presence anymore. (not that there were any)

Reply:
I've added a commit that makes sure this is happening.